### ISSUE TITLE
Missing API documentation for Native

### ISSUE THREAD

**Issue by Ajedi32** [opal/opal#1027]
Opal is currently missing API documentation for the `Native` class. This should be added.

----

This issue was created in response to the following conversation in issue #1024:

@ajedi32:
> My guess is you'd have to use Hash#to_native or #to_n or something? That doesn't seem to be documented anywhere though either, is it?

@ajedi32:
>Maybe some RDoc comments could be added to https://github.com/opal/opal/blob/master/stdlib/native.rb#L489-L520? Then we'd just need to use them to generate HTML and host them them somewhere. Better docs for some of the Opal-specific classes and methods / method implementations would be great.

@elia:
> yes, actually there's already a place for that kind of docs at http://opalrb.org/docs/ under "API documentation". It's unfortunate that for some reason Native isn't covered and we need to fix that.
>
> As of the array to_n should call to_n on all its elements, if it doesn't let me know and we'll ask @meh to fix it :blush:

**Comment by ashes999** [comment]
To clarify the last comment cited by @elia above, I have code like:

```
hash = { 
  :images => ['background.png', 'player.png'],
  :audio => ['beep.ogg']
}
```

`hash.to_n` gives me back a hash object that doesn't include the Opal bridge properties (around ~300 properties, mostly starting with `$`).  However, the two arrays I get back still do have those same Opal properties.

This caused problems in #1024, because CraftyJS iterates via `for x in hash`. (They support a number of different cases for their preloader, including sub-hashes, but anyway, they worked around the problem on their side.)

Please fix this. It's simple enough to diagnose with something like this (using the hash above):

```
hash = ...
hash = hash.to_n

`
var count = 0;
for (x in hash['images']) {
  count++;
}
console.log("Found " + count + " properties in the images array.");
`
```

You should only get a count of `2`, not `300`.

**Comment by Ajedi32** [comment]
This particular issue is about adding API *documentation* for the `Native` class. Fixing any incorrect behavior is a related but separate issue, IMO.

**Comment by ashes999** [comment]
Okay, I've created #1028 .

**Comment by elia** [comment]
Ok, so we now have API docs divided per version and per group (lib,corelib,stdlib)

http://opalrb.org/docs/api/v0.8.0/stdlib/index.html

Still lacks the actual documentation comments, I'd be happy to help out anyone who wants to takle this with a PR :smile: 

cc @meh

**Comment by elia** [comment]
closed by 74f2a8a7c7 (docs: Add some API documentation for native.rb)

It's not complete by any means, but I tried to cover the most important stuff.
